recmetrics/metrics.py

```diff
@@ -139,6 +139,37 @@ def _ark(actual: list, predicted: list, k=10) -> int:
 
     return score / len(actual)
 
+def _apk(actual: list, predicted: list, k=10) -> int:
+    """
+    Computes the average precision at k.
+    Parameters
+    ----------
+    actual : list
+        A list of actual items to be predicted
+    predicted : list
+        An ordered list of predicted items
+    k : int, default = 10
+        Number of predictions to consider
+    Returns:
+    -------
+    score : int
+        The average precision at k.
+    """
+
+    if len(predicted)>k:
+        predicted = predicted[:k]
+
+    if not predicted:
+        return 0.0
+
+    ## Precision is the number of true positives
+    ##  i.e. the number of things we predict that are
+    ##       actually in `actual`
+    ##  divided by the number of things we predicted
+    precision = sum([item in actual for item in set(predicted)]) / len(predicted)
```

Comment:
note to self: leave a comment here about why it's safe to divide by the number of predictions